pandas/io/sql.py

```diff
@@ -966,16 +971,35 @@ def _sqlalchemy_type(self, col):
             TIMESTAMP,
         )
 
-        if col_type == "datetime64" or col_type == "datetime":
+        if col_type in ("datetime64", "datetime", "datetimetz"):
             # GH 9086: TIMESTAMP is the suggested type if the column contains
             # timezone information
             try:
                 if col.dt.tz is not None:
                     return TIMESTAMP(timezone=True)
             except AttributeError:
-                # The column is actually a DatetimeIndex
-                if col.tz is not None:
-                    return TIMESTAMP(timezone=True)
+                # The column might be a DatetimeIndex
+                try:
+                    if col.tz is not None:
+                        return TIMESTAMP(timezone=True)
+                # or in the case of different offsets/timezones
+                # raise a better error message than
+                # "'Series' object has no attribute 'tz'"
+                except AttributeError as e:
+                    get_offset = lambda dt: (
```

Comment:
I think we can only hit this path if we have multiple timezones right? In that case I'd just re-raise a `ValueError` here and skip all this inference.